src/components/NewPost/NewPost.jsx

```diff
@@ -1,17 +1,41 @@
+import { useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { createPost } from "../../reducers/postSlice";
+
 export const NewPost = () => {
+  const [postData, setPostData] = useState({
+    content: "",
+  });
+  const { content } = postData;
+  const dispatch = useDispatch();
+  const { token } = useSelector((state) => state.auth);
+  const postHandler = (e) => {
+    setPostData({ ...postData, content: e.target.value });
+  };
+  const postSubmitHandler = () => {
+    dispatch(createPost({ postData, token }));
```

Comment:
need to check whether postData content is empty or has some value. User can create empty post

Reply:
Thanks Rahul for pointing this out, I have fixed it now.